internal/service/models/exceptions/userValidationErr.go

```diff
@@ -1,9 +1,9 @@
 package exceptions
 
 type UserValidationErr struct {
```

Comment:
This struct shouldn't be necessary. Instead we could just define all errors such as
var (
    ErrEmailIsRequired = errors.New("Email is required")
)
in there own file. We would then reference the ErrEmailIsRequired  variable when returning an error.

Reply:
This struct is necessary so Go can interpret that this Structs extends Error, is something like in Java UserValidationException extends RuntimeException

Reply:
Similar to my prior comment we don't use inheritance with except8ions in go. For custom errors we should create a generic struct that has an error message and error code. For an example of this see https://github.com/wexinc/ps-tag-payment-scheduler/blob/a6a283834641399513ba31f31bac27fda8aa7279/internal/domain/payment/models/errors.go#L8